src/TitleDetails/AddPieceModal/AddPieceModal.js

```diff
@@ -0,0 +1,206 @@
+      </Pluggable>
+      <Button
+        data-test-add-piece-cancel
+        onClick={close}
+      >
+        <FormattedMessage id="ui-receiving.piece.actions.cancel" />
+      </Button>
+    </ModalFooter>
+  );
+
+  return (
+    <Modal
+      enforceFocus={false}
+      footer={footer}
+      id="add-piece-modal"
+      label={<FormattedMessage id={labelId} />}
+      open
+    >
+      <form>
+        <Row>
+          <Col xs={6}>
+            <Field
+              component={TextField}
+              fullWidth
+              label={<FormattedMessage id="ui-receiving.piece.caption" />}
+              name="caption"
+              required
+              type="text"
+              validate={validateRequired}
+            />
+          </Col>
+          <Col xs>
+            <FieldSelectFinal
+              dataOptions={pieceFormatOptions}
+              label={<FormattedMessage id="ui-receiving.piece.format" />}
+              name="format"
+              required
+            />
+          </Col>
+        </Row>
+        <Row>
+          <Col xs>
+            <FieldDatepickerFinal
+              labelId="ui-receiving.piece.receiptDate"
+              name="receiptDate"
+              required
+              validate={validateRequired}
+            />
+          </Col>
+          <Col xs>
+            <Field
+              component={TextArea}
+              fullWidth
+              label={<FormattedMessage id="ui-receiving.piece.comment" />}
+              name="comment"
+            />
+          </Col>
+        </Row>
+        <Row>
+          <Col xs>
+            <FieldSelectFinal
+              dataOptions={locations}
+              fullWidth
+              label={<FormattedMessage id="ui-receiving.piece.location" />}
+              name="locationId"
+              required={isLocationRequired}
+            />
+          </Col>
+          <Col xs>
+            <Field
+              component={Checkbox}
+              fullWidth
+              label={<FormattedMessage id="ui-receiving.piece.supplement" />}
+              name="supplement"
+              type="checkbox"
+              vertical
+            />
+          </Col>
+        </Row>
+      </form>
+    </Modal>
+  );
+};
+
+AddPieceModal.propTypes = {
+  close: PropTypes.func.isRequired,
+  createInventoryValues: PropTypes.object.isRequired,
+  handleSubmit: PropTypes.func.isRequired,
+  locations: PropTypes.arrayOf(PropTypes.shape({
+    label: PropTypes.string.isRequired,
+    value: PropTypes.string.isRequired,
+  })),
+  form: PropTypes.object,
+  instanceId: PropTypes.string,
+  onCheckIn: PropTypes.func.isRequired,
+  stripes: stripesShape.isRequired,
+  pieceFormatOptions: PropTypes.arrayOf(PropTypes.shape({
+    label: PropTypes.string.isRequired,
+    value: PropTypes.string.isRequired,
+  })),
+};
+
+AddPieceModal.defaultProps = {
+  locations: [],
+  pieceFormatOptions: [],
+};
+
+export default stripesConnect(stripesFinalForm({
```

Comment:
Could you please move stripesConnect inside ui-plugin-create-item? thanks

Reply:
also, there is `withStripes` HOC to include just the `stripes` GOD-object

Reply:
for now used `withStripes`